Fail to compile

I can't compile the project. Here is the output:
> /tmp/libcsptr/check/test/utils.h: In function ‘make_test_case’:
> /tmp/libcsptr/check/test/utils.h:26:28: error: passing argument 2 of ‘_tcase_add_test’ from incompatible pointer type [-Werror=incompatible-pointer-types]
>    26 |         tcase_add_test(tc, *f);
>       |                            ^
>       |                            |
>       |                            TFun {aka void (*)(int)}
> /usr/include/check.h:331:69: note: expected ‘const TTest *’ {aka ‘const struct TTest *’} but argument is of type ‘TFun’ {aka ‘void (*)(int)’}
>   331 | CK_DLL_EXP void CK_EXPORT _tcase_add_test(TCase * tc, const TTest * ttest,
>       |                                                       ~~~~~~~~~~~~~~^~~~~
> cc1: all warnings being treated as errors

However without libcheck installed, I can compile without problem

--- a/check/test/array.c
+++ b/check/test/array.c
@@ -51,10 +51,11 @@ START_TEST (test_array_dtor_run_with_meta) {
     ck_assert_msg(dtor_run, "Expected destructor to run");
 } END_TEST
 
-TFun array_tests[] = {
-    test_array,
-    test_array_meta,
-    test_array_dtor_run,
-    test_array_dtor_run_with_meta,
-    NULL
-};
+TCase *make_array_tests(void) {
+    TCase *tc = tcase_create("array");
+    tcase_add_test(tc, test_array);
+    tcase_add_test(tc, test_array_meta);
+    tcase_add_test(tc, test_array_dtor_run);
+    tcase_add_test(tc, test_array_dtor_run_with_meta);
+    return tc;
+}

--- a/check/test/misc.c
+++ b/check/test/misc.c
@@ -20,10 +20,11 @@ START_TEST (test_alloc_failure) {
 } END_TEST
 #endif
 
-TFun misc_tests[] = {
+TCase *make_misc_tests(void) {
+    TCase *tc = tcase_create("misc");
 #ifndef SMALLOC_FIXED_ALLOCATOR
-    test_alloc_failure,
+    tcase_add_test(tc, test_alloc_failure);
 #endif
-    test_zero_size,
-    NULL
-};
+    tcase_add_test(tc, test_zero_size);
+    return tc;
+}

--- a/check/test/scalar.c
+++ b/check/test/scalar.c
@@ -36,11 +36,11 @@ START_TEST (test_dtor_run_with_meta) {
     ck_assert_msg(dtor_run, "Expected destructor to run");
 } END_TEST
 
-TFun scalar_tests[] = {
-    test_pointer_valid,
-    test_meta,
-    test_dtor_run,
-    test_dtor_run_with_meta,
-    NULL
-};
-
+TCase *make_scalar_tests(void) {
+    TCase *tc = tcase_create("scalar");
+    tcase_add_test(tc, test_pointer_valid);
+    tcase_add_test(tc, test_meta);
+    tcase_add_test(tc, test_dtor_run);
+    tcase_add_test(tc, test_dtor_run_with_meta);
+    return tc;
+}

--- a/check/test/shared.c
+++ b/check/test/shared.c
@@ -21,8 +21,9 @@ START_TEST (test_shared_sref) {
     ck_assert_msg(dtor_run == 0, "Expected destructor NOT to have run.");
 } END_TEST
 
-TFun shared_tests[] = {
-    test_shared_init,
-    test_shared_sref,
-    NULL
-};
+TCase *make_shared_tests(void) {
+    TCase *tc = tcase_create("shared");
+    tcase_add_test(tc, test_shared_init);
+    tcase_add_test(tc, test_shared_sref);
+    return tc;
+}

--- a/check/test/test.c
+++ b/check/test/test.c
@@ -9,10 +9,10 @@ const struct meta m = {1, 2, 3};
 
 Suite *master_suite(void) {
     Suite *s = suite_create("master");
-    suite_add_tcase(s, make_test_case("misc", misc_tests));
-    suite_add_tcase(s, make_test_case("scalar", scalar_tests));
-    suite_add_tcase(s, make_test_case("array", array_tests));
-    suite_add_tcase(s, make_test_case("shared", shared_tests));
+    suite_add_tcase(s, make_misc_tests());
+    suite_add_tcase(s, make_scalar_tests());
+    suite_add_tcase(s, make_array_tests());
+    suite_add_tcase(s, make_shared_tests());
     return s;
 }
 

--- a/check/test/test.h
+++ b/check/test/test.h
@@ -5,9 +5,9 @@
 
 Suite *unique_ptr_suite(void);
 
-extern TFun scalar_tests[];
-extern TFun array_tests[];
-extern TFun misc_tests[];
-extern TFun shared_tests[];
+TCase *make_scalar_tests(void);
+TCase *make_array_tests(void);
+TCase *make_misc_tests(void);
+TCase *make_shared_tests(void);
 
 #endif /* !TEST_H_ */

--- a/check/test/utils.h
+++ b/check/test/utils.h
@@ -19,14 +19,6 @@ inline void assert_valid_ptr(void *ptr) {
             "Expected unique_ptr to return an aligned pointer.");
 }
 
-__attribute__((always_inline))
-inline TCase *make_test_case(const char *name, TFun func[]) {
-    TCase *tc = tcase_create(name);
-    for (TFun *f = func; *f != NULL; ++f)
-        tcase_add_test(tc, *f);
-    return tc;
-}
-
 struct meta {
     int i;
     long l;

